### ISSUE TITLE
Clash of Property Name in IOperation

### ISSUE THREAD

**Issue by mahald** [ChilliCream/hotchocolate#2122]
**Describe the bug**
I'm testing your StrawberryShake Client.

There is currently a show-stopper that makes it impossible for me to use it:

The Property "Name" directly in IOperation is properly a bad idea as this clashes if Name is used a graphql query.
As this is very common the Name Property should not be directly in IOperation (or it should be called _Name).
I can't find any workaround and 80% of my "objects" got a name property.

**To Reproduce**
Here Code that was generated: As you See I have a property Name and this clashes with the Name Property of IOperation
```
namespace xxxAPI
{
    [System.CodeDom.Compiler.GeneratedCode("StrawberryShake", "11.0.0")]
    public partial class xxxOperation
        : IOperation<IxxxAdd>
    {
        public string Name => "xxxAdd";
        public IDocument Document => All.Default;
        public OperationKind Kind => OperationKind.Mutation;
        public Type ResultType => typeof(IxxxAdd);
        public Optional<string> Name { get; set; }
        public Optional<string> Description { get; set; }
        public Optional<string> ParentId { get; set; }


        public IReadOnlyList<VariableValue> GetVariableValues()
        {
            var variables = new List<VariableValue>();

            if (Name.HasValue)
            {
                variables.Add(new VariableValue("name", "String", Name.Value));
            }

            if (Description.HasValue)
            {
                variables.Add(new VariableValue("description", "String", Description.Value));
            }

            if (ParentId.HasValue)
            {
                variables.Add(new VariableValue("parentId", "ID", ParentId.Value));
            }

            return variables;
        }
    }
}
```

Here the part in the .graphql:
```
mutation xAdd($_name: String!, $description: String, $parentId: ID!){
    xxxAdd(name: $name, description: $description, parentId: $parentId){
        description
        id
        name
    }
}
```

**Expected behavior**
No "Name" property inside IOperation either put in a Class or use _Name for internal property.

Like so maybe?
```

namespace xxxAPI
    {

        public class OperationData
        { // defied somewhere else ...
            public string Name;
            public IDocument Document;
            public OperationKind Kind;
            public Type ResultType;
        }

        [System.CodeDom.Compiler.GeneratedCode("StrawberryShake", "11.0.0")]
        public partial class xxxOperation
            : IOperation<IxxxAdd>
        {
            public OperationData OperationData => new OperationData()
            {
                Name = "xxx",
                Document = All.Default,
                Kind = OperationKind.Mutation,
                ResultType = typeof(xxx)
            };
            public Optional<string> Name { get; set; }
            public Optional<string> Description { get; set; }
            public Optional<string> ParentId { get; set; }
```

**Desktop (please complete the following information):**
 - OS: Win10
 - Version preview 137 (138 can't be installed)

**Comment by michaelstaib** [comment]
This issue is known and we are working on this. At the moment the fix is blocked by another PR.

I will close this issue since we already have this fixed in our generator PR.

**Comment by michaelstaib** [comment]
https://github.com/ChilliCream/hotchocolate/pull/1487
